django/contrib/auth/views.py

```diff
@@ -438,20 +442,35 @@ class PasswordResetConfirmView(PasswordContextMixin, FormView):
     title = _('Enter new password')
     token_generator = default_token_generator
 
+    def _build_redirect_url(self):
+        current_token = self.request.resolver_match.kwargs['token']
+        return self.request.path.replace(current_token, INTERNAL_RESET_URL_TOKEN)
```

Comment:
I don't really see the advantage of going through the reverse and resolver_match logic, instead of this url token change.